content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,704 @@
+   ```
+   Recall that we defined the parameters in the material SRG in `VegetationBending_Common.azsli`. That's how we are able to reference them with `MaterialSrg` here.
+1. Repeat step 2 for the depth pass, `VegetationBending_DepthPass.azsli`.
+1. Save your files and open the Material Editor.
+1. Select the VegetationBending material that you made previously, (`my_material`), and find **Vegetation Bending** in the **Inspector** on the right. Adjust the x and y offsets as you see fit!
+1. Save and return to the Editor.
+1. Observe how the offset matches your inputs from the Material Editor!
+
+{{< image-width src="/images/atom-guide/vegetation-bending-tutorial/parameteroffset.png" width="100%" alt="Parameter offset added." >}}
+
+Congrats! Now you have taken the first step to writing your own custom shaders.
+
+## Prepare to add vegetation bending
+Now that we've edited our vertex shader, let's take the next step and prepare to add vegetation bending by adding an appropriate model.
+
+### Add a tree
+In order to test the code that we are about to write, we need an appropriate mesh!
+
+1. Open the **Editor** to your project and level. 
+1. In the **Entity Outliner**, **right-click** and select **Create entity**. **Right-click** the new entity in the **Entity Outliner**, and rename the entity to `Tree`.
+1. In the **Entity Inspector**, click **Add Component** and select **Mesh**.
+1. In the mesh component, under **Model Asset**, type `tree.fbx` or click the file button to find `tree.fbx` in your project's `Objects` folder.
+1. Again in the mesh component, click **Add Material Component**, click the file icon next to *Default Material* and select your material that you made earlier (`my_material`).
+
+Now, we have a tree (at an offset)! This tree is important because it uses vertex colors that we will be using to determine how the tree should bend.
+
+Note on vertex colors: You can use a DCC tool to color the vertices to match the type of bending you want to do on each part of the tree: 
+* Red vertex color: smaller movement with high frequency of random sinusoidal noise.
+* Green vertex color: delays the start of the movement for variations with high frequency of random sinusoidal noise.
+* Blue vertex color: larger movement and bending with low frequency of sinusoidal noise. 
+In the case of our tree, the trunk's vertices are colored blue and leaves are colored red.
+
+### Add a shader option
+Our tree mesh that we just added has colored vertices; however, not every mesh that you want to apply your material type to may have colored vertices. Therefore, we want to add a [*shader option*](/docs/atom-guide/dev-guide/shaders/azsl/#shader-variant-options) that allows the vertex shader to handle both of these cases. We also want to add the reference to the colors so we can use them in the code.
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+1. At the bottom of the file, add:
```

Comment:
- Consolidate steps
- This suggestion uses the file name only. But more importantly, you should be consistent. So only do this if you're also updating the file paths to use the file name only (per my earlier comment). 

```suggestion
1. In the `VegetationBending_Common.azsli` file, at the bottom, define a boolean shader option:
```